### ISSUE TITLE
Bind multiple addresses/interfaces to API and Gateway

### ISSUE THREAD

**Issue by BrendanBenshoof** [ipfs/go-ipfs#1649]
Currently I cannot bind the gateway and api to multiple addresses  and I am forced to use ipv4/0.0.0.0 OR ipv6/:: which means I can only bind them to one protocol or the other.

Suggested fix:
Allow a list of bind addresses for both API and Gateway

**Comment by jbenet** [comment]
:+1: to making these lists. need to:

- add support for list
- add to a migration  (dev0.4.0 or later, i say)

**Comment by Mikaela** [comment]
This has been fixed or at least I have my gateway and API on both 127.0.0.1 and ::1.

**Comment by Stebalien** [comment]
Yes, this has been fixed. The remaining issue is binding on both by default.
